### ISSUE TITLE
git fashion extension support

### ISSUE THREAD

**Issue by Kubuxu** [ipfs/go-ipfs#2309]
As user of many scripts it would be great to have extension/script support in main command.
It would work as `ipfs unknown -> ipfs-unknown`.

Proposed solution for now is to allow white-list unknown command to be passed to external script.

In future official extension list (in [weechat](https://weechat.org/scripts/) scripts fashion) would be really nice.

**Comment by whyrusleeping** [comment]
So we currently have `ipfs update` set up this way, but i purposefully didnt open it up fully. If we want, it should be pretty easy to add it in. @jbenet @diasdavid @RichardLitt thoughts?

**Comment by RichardLitt** [comment]
I don't understand. Can you point to an example elsewhere (is this like `man git-squash` or so, for instance)?

**Comment by Kubuxu** [comment]
You can in example install git add-on named `flow`. As its executable is named `git-flow` you can call its commands using: `git flow ...`.

**Comment by RichardLitt** [comment]
Ah. That sounds good to me. I also think listing out extensions somewhere (maybe on awesome-ipfs) makes sense.

**Comment by noffle** [comment]
@Kubuxu did you have a specific use case in mind?

**Comment by Kubuxu** [comment]
Yes, I have already quite a lot of scripts flowing around and it would be much simpler to used them if they were under common name of ipfs (ipfs-size, ipfs-repin, ipfs-publish, ipfs-updateboot and so on).
